### PR TITLE
Add methods to Output

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add feature_blocks(), immutable_feature_blocks() and unlock_conditions() methods to Output

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bee-message/src/output/mod.rs [hunk 1]
```diff
@@ -138,6 +138,36 @@ impl Output {
         }
     }
 
+    /// Returns the feature blocks of an `Output`, if any.
+    pub fn feature_blocks(&self) -> Option<&[FeatureBlock]> {
```
Comment (thibault-martinez) [inline bee-message/src/output/mod.rs]: Can you order these methods please ? NativeTokens, UnlockConditions, FeatureBlocks, ImmutableFeatureBlocks

Comment (Thoralf-M) [reply bee-message/src/output/mod.rs]: I ordered them alphabetically :cry:

Comment (thibault-martinez) [reply bee-message/src/output/mod.rs]: Need to be ordered by fields order :)

**Review by samuel-rufi — DISMISSED** [review]

File: bee-message/src/output/mod.rs [hunk 2]
```diff
@@ -152,4 +152,49 @@ impl Output {
             Self::Nft(output) => Some(output.native_tokens()),
         }
     }
+
+    /// Returns the unlock conditions of an `Output`, if any.
+    pub fn unlock_conditions(&self) -> Option<&[UnlockCondition]> {
```
Comment (thibault-martinez) [inline bee-message/src/output/mod.rs]: Can you return containers instead of slices for consistency ?

**Review by thibault-martinez — APPROVED** [review]

**Review by Adam-Gleave — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**